README.md

```diff
@@ -1,4 +1,146 @@
+    - `libs/` - external JS code such as Mustache
+    - `modules/` - the modular UI blocks (see below)
+    - `widgets/` - some GTK widgets for UI code that are not modular UI blocks
+  - `search/` - library for accessing databases of offline content
+  - `tools/` - various testing and development tools
+- `lib/` - C library code
+  - `eosknowledgeprivate/` - introspectable C wrappers for `js/app/`
+  - `eosknowledgesearchprivate/` - introspectable C wrappers for `js/search/`
+  - `web-extensions/` - WebKit web extensions for use when viewing documents
+- `m4/` - Autoconf code
+- `po/` - internationalization (see below)
+- `tests/` - unit and integration tests
+  - Various utility code for tests
+  - `autobahn/` - integration tests for the Autobahn YAML processor
+  - `js/` - unit tests for the `js/app` and `js/search` libraries
+  - `test-content/` - sample content for some tests
+
+Modular UI blocks
+-----------------
+UI modules are GObjects, with a few extra features and a few restrictions.
+They are meant to be assembled in a declarative file, without writing having to write any code such as signal handlers.
+
+First of all there are no signals, because there is no code.
+(A UI module's behavior is baked into its code.)
+All properties accessible from the declarative file are treated as construct-only (except when using property bindings.)
+Modules have **slots** and **references**, which are named spots where other modules can be contained or referenced.
+
+Here are the types of UI modules and how they fit together.
+See `js/app/modules/` for the code.
+
+- **Arrangement** - These modules generally have a `card` slot which a **Card** can go into.
+  They arrange a set of dynamically created cards on the screen.
+- **Banner** - These modules represent a piece of data which is presented to the user, such as an app's logo or a search term.
+- **Card** - These modules represent a document in the database of offline content.
+  They display its metadata (title, keywords) in various ways.
+- **ContentGroup** - This is the heart of displaying offline content.
+  Content groups have `selection` and `arrangement` slots.
+  They make sure the **Arrangement** receives (and creates **Card**s for) the records that the **Selection** wants to display.
+- **Controller** - These modules define different types of app experiences.
+  They are in charge of what happens when you click somewhere.
+- **Decoration** - These modules are UI elements which are used for display only.
+- **Filter** - These modules can be added to a **Selection** to display only some of its records.
+- **Layout** - These modules contain submodules and arrange them on one page of the app.
+- **Navigation** - These modules are UI elements which are used for navigation through an app.
+- **Order** - These modules can be added to a **Selection** to change the sorting order in which its records are displayed.
+- **Pager** - Modules can also be pages of an app.
```

Comment:
You say you can arrange them "in the form of pages", but it still doesn't tell me what a page is. It says "only one at a time is displayed", but I didn't know if that meant "only one page is displayed" or "only one module is displayed" (or both?)

The sentence doesn't scan well for someone unfamiliar and confused with the subject matter.